### PR TITLE
Disable select text (#218)

### PR DESCRIPTION
**Pull request by abiduzz420**
**What does this PR solve ?**
  - It disables selecting text in the application (See #218)

**Why should this PR be considered?**
  - Having select text hampers the user experience of the application.

_If there any changes or modifications to be made with regards to code style guide or anything, do let me know. Thanks!_

### CODE REVIEW

**Comment by abiduzz420** [comment]
@mkermani144 Could you help me out in figuring out where the TravicCI failed ?

**Comment by abiduzz420** [comment]
Hi @mkermani144 I would like to know the issues which caused the TravisCI checks to fail. I am still very naive with TravisCI. I shall be happy to fix the errors :)

**Comment by mkermani144** [comment]
@abiduzz420 
Sorry for late response. I'm busy these days 😉.
I think your PR fails because the repo E2E tests can only be run from PRs of my own repo, not any forks. I will check your PR in a few days and if it's OK I will try to fix the failure or merge it anyway.

**Comment by abiduzz420** [comment]
Thank you @mkermani144 . Please let me know if you need any help.

**Comment by mkermani144** [comment]
@abiduzz420 
I'm so sorry for my very late response. I fixed the problem that caused Travis to fail (see #253) and then merged the new master branch into yours. Now all your PR checks pass and there is no problem anymore.
I will review your code asap, as it's not ready to be merged into the master branch and does need some major changes.

**Comment by abiduzz420** [comment]
Okay sure @mkermani144 Do let me know if you need anything.

File: src/App.css [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
-  width: 100vw;
-  height: 100vh;
-  overflow-x: hidden;
-  background-color: #FAFAFA;
+	display: flex;
+	flex-direction: column;
+	width: 100vw;
+	height: 100vh;
+	overflow-x: hidden;
+	background-color: #fafafa;
```
Comment (mkermani144) [inline src/App.css]: Should not change hex numbers to lowercase

File: src/App.css [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
-  display: flex;
-  flex-direction: row;
+	flex: 1;
+	display: flex;
+	flex-direction: row;
 }
 .AppBar {
-  position: fixed !important;
+	position: fixed !important;
+	-moz-user-select: none; /* Firefox */
```
Comment (mkermani144) [inline src/App.css]: There is no need to support all browsers, as this is an electron app and `user-select: none;` will suffice.

File: src/FAB/NewIdeaDialog.js [hunk 3]
```diff
@@ -52,6 +54,11 @@ class NewIdeaDialog extends Component {
     const dialogTitleStyle = {
       backgroundColor: yellow800,
       color: grey50,
+      MozUserSelect: 'none', /* Firefox */
```
Comment (mkermani144) [inline src/FAB/NewIdeaDialog.js]: Again, redundant browser support, while this is an electron app

File: src/Task/TaskList.js [hunk 4]
```diff
@@ -167,8 +187,8 @@ class TaskList extends Component {
               <Task
                 color={task.color}
                 signal={task.signal}
-                text={task.task}
                 estimation={task.estimation}
+                text={task.task}
```
Comment (mkermani144) [inline src/Task/TaskList.js]: Redundant changing the line of `text={task.task}`

File: src/Idea/IdeaList.css [hunk 5]
```diff
@@ -1,36 +1,36 @@
...
+	cursor: default;
+	user-select: none;
 }
 .ideas-empty-state h1 {
-  font-size: 500%;
-  margin-bottom: 0;
-  opacity: .2;
+	font-size: 500%;
+	margin-bottom: 0;
+	opacity: 0.2;
```
Comment (mkermani144) [inline src/Idea/IdeaList.css]: Extra zero not needed

**Comment by mkermani144** [comment]
In addition to my comments, I think using a global CSS style (not a JS one) such as
```css
* {
  user-select: none;
}
```
in `App.css` will do the job without any extra effort and multiple file changes, as this is an app and no user ever need to select any text in the app most of the times (except for inputs and text areas). In addition, it adapts better to the current state of the repo, as we use CSS styles wherever possible, and switch to JS ones only when we have to. So I suggest you close this PR and make another one to prevent adding any extra commits to the repo.

**Comment by mkermani144** [comment]
@abiduzz420 
Please let me know if you are interested in working on #218 anymore, as I'm going to release a new version of the app and #218 needs to be fixed before the release.

**Comment by abiduzz420** [comment]
Hi @mkermani144 I have a bit busy due to my exams, can I get this fixed by Sunday ?

**Comment by mkermani144** [comment]
@abiduzz420
Certainly. I can wait until the end of Sunday. Please don't forget to consider the points I mentioned in my previous comment, so I can merge your next PR fast and without any change requests.

**Comment by mkermani144** [comment]
Because of no response, I close this PR. To fix the issue I opened #264.

**Pull request closed without merging**